### PR TITLE
feat: add `parityCheckStatus` field to `array` query

### PR DESCRIPTION
**Pull request by pujitm**
Responds with a ParityCheck:
```ts
type ParityCheck {
  """Date of the parity check"""
  date: DateTime

  """Duration of the parity check in seconds"""
  duration: Int

  """Speed of the parity check, in MB/s"""
  speed: String

  """Status of the parity check"""
  status: ParityCheckStatus!

  """Number of errors during the parity check"""
  errors: Int

  """Progress percentage of the parity check"""
  progress: Int

  """Whether corrections are being written to parity"""
  correcting: Boolean

  """Whether the parity check is paused"""
  paused: Boolean

  """Whether the parity check is running"""
  running: Boolean
}

enum ParityCheckStatus {
    NEVER_RUN = 'never_run',
    RUNNING = 'running',
    PAUSED = 'paused',
    COMPLETED = 'completed',
    CANCELLED = 'cancelled',
    FAILED = 'failed',
}
```

## Summary by CodeRabbit

- New Features
  - Exposes a structured parity-check status for arrays with detailed fields (status enum: NEVER_RUN, RUNNING, PAUSED, COMPLETED, CANCELLED, FAILED), date, duration, speed, progress, errors, and running/paused flags.

- Tests
  - Adds comprehensive unit tests covering all parity-check states, numeric edge cases, speed/date/duration/progress calculations, and real-world scenarios.

- Refactor
  - Safer numeric/date parsing and a new numeric-conversion helper; minor formatting/cleanup in shared utilities.

### CODE REVIEW

File: api/src/core/modules/array/get-array-data.ts [hunk 1]
```diff
@@ -61,5 +62,6 @@ export const getArrayData = (getState = store.getState): UnraidArray => {
         parities,
         disks,
         caches,
+        parityCheckStatus: getParityCheckStatus(emhttp.var),
```
Comment (elibosley) [inline api/src/core/modules/array/get-array-data.ts:65]: I'd really ideally like to see this be an object containing a status and a percentage, so it's more usable to an end user than just "running" or not.
